### PR TITLE
[DUOS-1595][risk=no] consent-update-get-dar-collection-by-id-to-allow-for-other-roles

### PR DESCRIPTION
**Pull request by shaemarks**
[Link to Jira ticket](https://broadworkbench.atlassian.net/browse/DUOS-1595)
Summary of changes:

- Added logic to getDarCollectionById to allow users that did not create the dar to still get access to it if their role permissions meet certain criteria 
- Added unit tests to DarCollectionResourceTest for each role

----
Have you read [CONTRIBUTING.md](../CONTRIBUTING.md) lately? If not, do that first.

- Label PR with a Jira ticket number and include a link to the ticket
- Label PR with a security risk modifier [no, low, medium, high]
- PR describes scope of changes
- Get a minimum of one thumbs worth of review, preferably two if enough team members are available
- Get PO sign-off for all non-trivial UI or workflow changes
- Verify all tests go green
- Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: src/main/java/org/broadinstitute/consent/http/resources/DarCollectionResource.java [hunk 1]
```diff
@@ -5,12 +5,7 @@
 import io.dropwizard.auth.Auth;
 import org.broadinstitute.consent.http.enumeration.DarStatus;
 import org.broadinstitute.consent.http.enumeration.UserRoles;
-import org.broadinstitute.consent.http.models.AuthUser;
-import org.broadinstitute.consent.http.models.DarCollection;
-import org.broadinstitute.consent.http.models.DataAccessRequest;
-import org.broadinstitute.consent.http.models.PaginationResponse;
-import org.broadinstitute.consent.http.models.PaginationToken;
-import org.broadinstitute.consent.http.models.User;
+import org.broadinstitute.consent.http.models.*;
```
Comment (rushtong) [inline src/main/java/org/broadinstitute/consent/http/resources/DarCollectionResource.java]: We don't use star imports. Look for the setting in intellij that lets you set it to a very large number (i.e, 9999).
Screen Shot 2022-02-07 at 1 18 27 PM

File: src/test/java/org/broadinstitute/consent/http/resources/DarCollectionResourceTest.java [hunk 2]
```diff
@@ -24,14 +24,7 @@
...
 import javax.ws.rs.core.Response;
-import java.util.ArrayList;
-import java.util.Collections;
-import java.util.Date;
-import java.util.HashSet;
-import java.util.List;
-import java.util.Map;
-import java.util.Set;
-import java.util.UUID;
+import java.util.*;
```
Comment (rushtong) [inline src/test/java/org/broadinstitute/consent/http/resources/DarCollectionResourceTest.java]: Same star-import issue here.

File: src/test/java/org/broadinstitute/consent/http/resources/DarCollectionResourceTest.java [hunk 3]
```diff
@@ -47,6 +40,9 @@
     new UserRole(UserRoles.RESEARCHER.getRoleId(), UserRoles.RESEARCHER.getRoleName())
   );
   private final User researcher = new User(1, authUser.getEmail(), "Display Name", new Date(), researcherRole, authUser.getEmail());
+  private final List<UserRole> signingOfficialRole = Collections.singletonList(
```
Comment (rushtong) [inline src/test/java/org/broadinstitute/consent/http/resources/DarCollectionResourceTest.java]: `List.of` is a little easier to read. Was added in later versions of java.

```suggestion
  private final List<UserRole> signingOfficialRole = List.of(
```

**Review by rushtong — COMMENTED** [review]
See comments inline, code looks good!

**Review by JVThomas — APPROVED** [review]
Functionally the code looks great and the tests extensively cover the role variances well. That said, I'm not an expert on Java syntax so I can't provide a good insight into that.

**Review by rushtong — APPROVED** [review]
👍🏽

**Pull request merged**